docs/source/processes.rst

```diff
@@ -770,13 +778,56 @@ described in |pywps-multi-output|_.
 .. seealso::
     - :ref:`Multiple and Optional Values`
 
+.. _exec_output_location:
+
+Outputs Location
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, :term:`Job` results will be hosted under the endpoint configured by ``weaver.wps_output_url`` and
+``weaver.wps_output_path``, and will be stored under directory defined by ``weaver.wps_output_dir`` setting.
+
+Each :term:`Job` will have its specific UUID employed for all of the outputs files, logs and status in order to
+avoid conflicts. Therefore, outputs will be available with the following location:
+
+.. code-block::
+
+    {WPS_OUTPUT_URL}/{JOB_UUID}.xml             # status location
+    {WPS_OUTPUT_URL}/{JOB_UUID}.log             # execution logs
+    {WPS_OUTPUT_URL}/{JOB_UUID}/{output.ext}    # results of the job if successful
+
+.. note::
+    Value ``WPS_OUTPUT_URL`` in above example is resolved accordingly with ``weaver.wps_output_url``,
+    ``weaver.wps_output_path`` and ``weaver.url``, as per :ref:`conf_settings` details.
+
+When submitting a :term:`Job` for execution, it is also possible to provide the ``X-WPS-Output-Context`` header.
+This modifies the output location to be nested under the specified directory or sub-directories.
+
+For example, providing ``X-WPS-Output-Context: project/test-1`` will result in outputs located at
+
+.. code-block::
+
+    {WPS_OUTPUT_URL}/project/test-1/{JOB_UUID}/{output.ext}
+
+.. note::
+    Values provided by ``X-WPS-Output-Context`` can only contain alphanumeric, hyphens, underscores and path separators
+    that will result in a valid directory and URL locations. The path is assumed relative to the resolved :term:`WPS`
+    output directory, and will therefore reject any ``.`` or ``..`` path references.
+    The path must also **NOT** start by ``/``.
```

Comment:
Should be stripped properly I guess, else what? error?

Reply:
Yes, there is a check for this kind of values and corresponding test in the PR. 
I can add the detail explicitly that an error is raised in this case in the docs.

Reply:
Yeah I saw the error handling, but I was wondering why we cannot strip them or even handle ./ prefixes silently? But in the end having one way to do a thing is better given that it's well documented.

Reply:
I decided to raise instead of silently handling `.` because it is for one easier (especially for cases like `path/./path/ok`), but also because I explicitly forbid `..` and starting `/` to go up the tree / forge another base dir. It felt more confusing to support `.` but not the others.